mvpa2/generators/partition.py

```diff
@@ -491,6 +493,54 @@ def _get_partition_specs(self, uniqueattrs):
                     if ind < self.count]
 
 
+class FactorialPartitioner(Partitioner):
+    """
+    Given another partitioner on a dataset with two hierarchically organized
+    attributes, generates the combinations of all possible folds of the
+    superordinate attribute with the balanced combinations of the subordinate
+    attribute. For example, splitting familiar and unfamiliar faces while
+    balancing identities in the training/test set.
+
+    ADD EXAMPLE HERE
+    """
+    def __init__(self, partitioner, **kwargs):
+        Partitioner.__init__(self, **kwargs)
+        # store the subordinate partitioner
+        self.partitioner = partitioner
+
+    def generate(self, ds):
+        class FakeDataset(object):
+            def __init__(self, ds):
+                self.sa = ds.sa
+
+            def copy(self, deep=False):
+                assert(not deep)  # dunno how
+                ds = FakeDataset(self)
+                ds.sa = ds.sa.copy(deep=False)
+                return ds
+
+            def __getitem__(self, item):
+                return FakeDataset(self[item])
+
+        #fakeds = FakeDataset(ds)
```

Comment:
kill FakeDataset above since not used, and provide description under for why selecting a single feature